zap/src/main/java/org/parosproxy/paros/core/scanner/Analyser.java

```diff
@@ -514,11 +515,11 @@ private void sendAndReceive(HttpMessage msg) throws HttpException, IOException {
             }
         }
 
-        httpSender.sendAndReceive(msg, parent.getRedirectRequestConfig());
-        requestCount++;
-
-        // ZAP: Notify parent
         if (parent != null) {
+            httpSender.sendAndReceive(msg, parent.getRedirectRequestConfig());
+            requestCount++;
+
+            // ZAP: Notify parent
             parent.notifyNewMessage(msg);
```

Comment:
Shall I remove the blank and comment? (The comment seems unnecessary)

Reply:
Yes, we can also remove the parent null check, it will never happen here (it would break earlier).

Reply:
parent is instantiated null at L102.

Reply:
Yes, but it would fail in `analyse` not here (only the constructor with parent is being used, so the parent is never null anyway).